Sources/Overlays/_Testing_Foundation/Attachments/Attachment+URL.swift

```diff
@@ -83,79 +81,178 @@ extension Attachment where AttachableValue == Data {
-    let temporaryName = "\(UUID().uuidString).tgz"
-    let temporaryURL = FileManager.default.temporaryDirectory.appendingPathComponent(temporaryName)
+/// The path to the archiver tool.
+///
+/// This path refers to a file (named `_archiverName`) within the `"System32"`
+/// folder of the current system, which is not always located in `"C:\Windows."`
+///
+/// If the path cannot be determined, the value of this property is `nil`.
+private let _archiverPath: String? = {
+  let bufferCount = GetSystemDirectoryW(nil, 0)
+  guard bufferCount > 0 else {
+    return nil
+  }
+
+  return withUnsafeTemporaryAllocation(of: wchar_t.self, capacity: Int(bufferCount)) { buffer -> String? in
+    let bufferCount = GetSystemDirectoryW(buffer.baseAddress!, UINT(buffer.count))
+    guard bufferCount > 0 && bufferCount < buffer.count else {
+      return nil
+    }
 
+    return _archiverName.withCString(encodedAs: UTF16.self) { archiverName -> String? in
+      var result: UnsafeMutablePointer<wchar_t>?
+
+      let flags = ULONG(PATHCCH_ALLOW_LONG_PATHS.rawValue)
+      guard S_OK == PathAllocCombine(buffer.baseAddress!, archiverName, flags, &result) else {
+        return nil
+      }
+      defer {
+        LocalFree(result)
+      }
+
+      return result.flatMap { String.decodeCString($0, as: UTF16.self)?.result }
+    }
+  }
+}()
+#endif
+
+/// Compress the contents of a directory to an archive, then map that archive
+/// back into memory.
+///
+/// - Parameters:
+///   - directoryURL: A URL referring to the directory to attach.
+///
+/// - Returns: An instance of `Data` containing the compressed contents of the
+///   given directory.
+///
+/// - Throws: Any error encountered trying to compress the directory, or if
+///   directories cannot be compressed on this platform.
+///
+/// This function asynchronously compresses the contents of `directoryURL` into
+/// an archive (currently of `.zip` format, although this is subject to change.)
+private func compressContentsOfDirectory(at directoryURL: URL) async throws -> Data {
 #if !SWT_NO_PROCESS_SPAWNING
-#if os(Windows)
-    let tarPath = #"C:\Windows\System32\tar.exe"#
+  let temporaryName = "\(UUID().uuidString).zip"
+  let temporaryURL = FileManager.default.temporaryDirectory.appendingPathComponent(temporaryName)
+  defer {
+    try? FileManager().removeItem(at: temporaryURL)
+  }
+
+  // The standard version of tar(1) does not (appear to) support writing PKZIP
+  // archives. FreeBSD's (AKA bsdtar) was long ago rebased atop libarchive and
+  // knows how to write PKZIP archives, while Windows inherited FreeBSD's tar
+  // tool in Windows 10 Build 17063 (per https://techcommunity.microsoft.com/blog/containers/tar-and-curl-come-to-windows/382409).
+  //
+  // On Linux (which does not have FreeBSD's version of tar(1)), we can use
+  // zip(1) instead.
+#if os(Linux)
+  let archiverPath = "/usr/bin/zip"
+#elseif SWT_TARGET_OS_APPLE || os(FreeBSD)
+  let archiverPath = "/usr/bin/tar"
```

Comment:
I was under the impression this `compressContentsOfDirectory()` function was not needed on Apple platforms. Should we entirely compile it out `#if SWT_TARGET_OS_APPLE`?

Reply:
There may be variant Apple platforms (think some minimal OSS Darwin install?) that don't have file coordination. It's basically just a fallback implementation we can use if `NSFileCoordinator` isn't available.